optimum/intel/openvino/quantization.py

```diff
@@ -246,6 +256,42 @@ def _quantize_ovbasemodel(
         self.model.model = quantized_model
         self.model.save_pretrained(save_directory)
 
+    def _quantize_ovmodelforseq2seqlm(
+        self,
+        calibration_dataset: Dataset,
+        save_directory: Union[str, Path],
+        batch_size: int = 1,
+        data_collator: Optional[DataCollator] = None,
+        remove_unused_columns: bool = True,
+        **kwargs,
+    ):
+        save_directory = Path(save_directory)
+        save_directory.mkdir(parents=True, exist_ok=True)
+
+        calibration_dataloader = self._get_calibration_dataloader(
+            calibration_dataset=calibration_dataset,
+            batch_size=batch_size,
+            remove_unused_columns=remove_unused_columns,
+            data_collator=data_collator,
+        )
+        quantization_dataset = nncf.Dataset(calibration_dataloader, lambda x: x)
+
+        # Full quantization of encoder
+        self.model.encoder_model = nncf.quantize(
+            self.model.encoder_model,
+            quantization_dataset,
+            model_type=nncf.ModelType.TRANSFORMER,
+            fast_bias_correction=kwargs.get("fast_bias_correction", True),
+            **kwargs,
+        )
+
+        # Compress weights of decoders for safity
+        self.model.decoder_model = nncf.compress_weights(self.model.decoder_model)
+        if self.model.use_cache:
+            self.model.decoder_with_past_model = nncf.compress_weights(self.model.decoder_with_past_model)
```

Comment:
Would there be a possibility to also quantize the activations for the decoder components?

Reply:
It is possible but it may hurt accuracy.